### PR TITLE
Fix short 180 degree rotation when using set_bone_position

### LINKED ISSUE #10404: Bone rotations inverted on set_animation if using set_bone_position on the body
##### Summary

See [this video](https://youtu.be/K4xtXjgrAL0). The face flashes on the back of the head as animations are started (when beginning walking/digging). This applies not only to 3rd person mode but also to other observing clients. Minetest version: 5.4-dev

##### Steps to reproduce

```lua
minetest.register_globalstep(function() for _, player in pairs(minetest.get_connected_players()) do player:set_bone_position("Body", {x = 0, y = 6.5, z = 0}, {x = 0, y = 180, z = 0}) end end)
```

### PR DESCRIPTION
**Pull request by appgurueu**
Closes #10404 - Ready for Review. Reproduce the issue, confirm that this PR fixes it.

### CODE REVIEW

File: src/client/content_cao.cpp [hunk 1]
```diff
@@ -1422,24 +1422,17 @@ void GenericCAO::updateBonePosition()
 		if (!bone)
 			continue;
 
-		//If bone is manually positioned there is no need to perform the bug check
-		bool skip = false;
-		for (auto &it : m_bone_position) {
-			if (it.first == bone->getName()) {
```
Comment (sfan5) [inline src/client/content_cao.cpp:1428]: Why is it that manually positioned bones *do* need the workaround?

Comment (appgurueu) [reply src/client/content_cao.cpp:1428]: Empirical evidence suggests it.

**Review by sfan5 — APPROVED** [review]
Looks logical.

**Review by SmallJoker — APPROVED** [review]
Fixes the described issue.

File: src/client/content_cao.cpp (hunk 1 above)
Comment (SmallJoker) [reply src/client/content_cao.cpp:1428]: "Empirical evidence suggests it." I don't understand what you mean.
Well, I tested the same situation with this optimization step - and the glitch was still there.

**Pull request merged**